crates/rpc/rpc-eth-types/src/error/mod.rs

```diff
@@ -775,6 +775,9 @@ pub enum RpcPoolError {
     /// Custom pool error
     #[error(transparent)]
     PoolTransactionError(Box<dyn PoolTransactionError>),
+    /// EIP-2681 related error
+    #[error("nonce is not valid")]
+    Eip2681,
```

Comment:
we can get rid of this and instead simply map this to RpcInvalidTransactionError::NonceMaxValue

which is the variant that we already have for this